### PR TITLE
UX: Better composer hyperlink modal

### PR DESCRIPTION
**Pull request by pmusaraj**
This brings the hyperlink modal in line with all other modals (upload, date, etc.) by placing the fixed modal element outside of the composer. 

Before: 

image

After: 

image

### CODE REVIEW

File: app/assets/javascripts/discourse/controllers/insert-hyperlink.js.es6 [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import { default as computed } from "ember-addons/ember-computed-decorators";
+
+export default Ember.Controller.extend(ModalFunctionality, {
+  linkUrl: "",
+  linkText: "",
+
+  @computed()
+  title: () => "composer.link_dialog_title",
```
Comment (CvX) [inline app/assets/javascripts/discourse/controllers/insert-hyperlink.js.es6]: Does it have to be a computed property if the value is static?

Comment (pmusaraj) [reply app/assets/javascripts/discourse/controllers/insert-hyperlink.js.es6]: Good spot, no, it doesn't need to be computed at all.

File: app/assets/javascripts/discourse/controllers/insert-hyperlink.js.es6 [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import { default as computed } from "ember-addons/ember-computed-decorators";
```
Comment (CvX) [inline app/assets/javascripts/discourse/controllers/insert-hyperlink.js.es6]: ESLint error, `computed` is no longer used: https://travis-ci.org/discourse/discourse/jobs/593993307#L701-L702

**Comment by SamSaffron** [comment]
I support this cleanup, feel free to merge!

**Comment by CvX** [comment]
@pmusaraj It seems it's no longer possible to submit the form by pressing enter/return. Could this PR be the cause?

**Pull request merged**